app/src/main/java/it/niedermann/owncloud/notes/persistence/migration/Migration_21_22.java

```diff
@@ -0,0 +1,46 @@
+package it.niedermann.owncloud.notes.persistence.migration;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import androidx.annotation.NonNull;
+import androidx.preference.PreferenceManager;
+import androidx.room.migration.Migration;
+import androidx.sqlite.db.SupportSQLiteDatabase;
+
+import it.niedermann.owncloud.notes.persistence.SyncWorker;
+
+/**
+ * Enabling backgroundSync, set from {@link String} values to {@link Boolean} values
+ * https://github.com/stefan-niedermann/nextcloud-notes/issues/1168
+ */
+
+public class Migration_21_22 extends Migration {
+    @NonNull
+    private final Context context;
+
+    public Migration_21_22(@NonNull Context context) {
+        super(21, 22);
+        this.context = context;
+    }
+
+    @Override
+    public void migrate(@NonNull SupportSQLiteDatabase database) {
+        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+        SharedPreferences.Editor editor = sharedPreferences.edit();
+        if (sharedPreferences.contains("backgroundSync")) {
+            if (sharedPreferences.getString("backgroundSync", "on").equals("off")) {
```

Comment:
Please use `""` as default value instead of `"on"`. The default value should never be used anyway, because we already check with `.contains(...)` the line before, but we never used the value `"on"` and it will be confusing to read it here in the future.